crates/wit-parser/src/ast/expand.rs

```diff
@@ -0,0 +1,100 @@
+use anyhow::{anyhow, Result};
+use indexmap::IndexMap;
+use std::collections::{HashMap, HashSet};
+use crate::{Resolve, World, WorldId, PackageId, WorldItem, InterfaceId};
+
+pub type Substitutions = HashMap<String, HashMap<String, HashSet<String>>>;
+
+pub fn expand(
+    resolve: &mut Resolve,
+    package_id: PackageId,
+    world: Option<&str>,
+    substitutions: Substitutions
+) -> Result<()> {
+    let world_id = resolve.select_world(package_id, world)?;
+    resolve.expand_wildcards(world_id, substitutions)
+}
+
+impl Resolve {
+    pub fn expand_wildcards(&mut self, world_id: WorldId, substitutions: Substitutions) -> Result<()> {
```

Comment:
The top level of this map is the name of the `world_id` specified, so could one layer of the map be peeled off perhaps and the argument here is "just" `HashMap<String, T<String>>`?